### ISSUE TITLE
Support JDK 17

### ISSUE THREAD

**Issue by lazystone** [paketo-buildpacks/bellsoft-liberica#127]
## What happened?

* What were you attempting to do?

Trying to build an image using spring-boot/gradle under JDK 17

* What did you expect to happen?

The expected behaviour is an image being built.

* What was the actual behavior? Please provide log output, if possible.

```
> Task :bootBuildImage
Building image 'foo.bar/abc/qwerty:latest'
 > Pulling builder image 'docker.io/paketobuildpacks/builder:base' ..................................................
 > Pulled builder image 'paketobuildpacks/builder@sha256:624fcec632bf46a3cd72d95f760070455714c8685becb901a2f5ee7b6d856266'
 > Pulling run image 'docker.io/paketobuildpacks/run:base-cnb' ..................................................
 > Pulled run image 'paketobuildpacks/run@sha256:64e37b972ee65672ada65f2e49d2be0d0d47b87d868201965794b6c89fb6cd44'
 > Executing lifecycle version v0.11.4
 > Using build cache volume 'pack-cache-5e6a6dcf80bc.build'

 > Running creator
    [creator]     ===> DETECTING
    [creator]     5 of 18 buildpacks participating
    [creator]     paketo-buildpacks/ca-certificates   2.4.1
    [creator]     paketo-buildpacks/bellsoft-liberica 8.5.0
    [creator]     paketo-buildpacks/executable-jar    5.2.1
    [creator]     paketo-buildpacks/dist-zip          4.2.1
    [creator]     paketo-buildpacks/spring-boot       4.6.0
    [creator]     ===> ANALYZING
    [creator]     Previous image with name "foo.bar/abc/qwerty:latest" not found
    [creator]     ===> RESTORING
    [creator]     ===> BUILDING
    [creator]     
    [creator]     Paketo CA Certificates Buildpack 2.4.1
    [creator]       https://github.com/paketo-buildpacks/ca-certificates
    [creator]       Launch Helper: Contributing to layer
    [creator]         Creating /layers/paketo-buildpacks_ca-certificates/helper/exec.d/ca-certificates-helper
    [creator]     
    [creator]     Paketo BellSoft Liberica Buildpack 8.5.0
    [creator]       https://github.com/paketo-buildpacks/bellsoft-liberica
    [creator]       Build Configuration:
    [creator]         $BP_JVM_TYPE                 JRE             the JVM type - JDK or JRE
    [creator]         $BP_JVM_VERSION              17.*            the Java version
    [creator]       Launch Configuration:
    [creator]         $BPL_HEAP_DUMP_PATH                          write heap dumps on error to this path
    [creator]         $BPL_JVM_HEAD_ROOM           0               the headroom in memory calculation
    [creator]         $BPL_JVM_LOADED_CLASS_COUNT  35% of classes  the number of loaded classes in memory calculation
    [creator]         $BPL_JVM_THREAD_COUNT        250             the number of threads in memory calculation
    [creator]         $JAVA_TOOL_OPTIONS                           the JVM launch flags
    [creator]       No valid JRE available, providing matching JDK instead. Using a JDK at runtime has security implications.
    [creator]     
    [creator]     Paketo BellSoft Liberica Buildpack 8.5.0
    [creator]       unable to find dependency
    [creator]       no valid dependencies for jdk, 17.*, and io.buildpacks.stacks.bionic in [(jdk, 8.0.302, [io.buildpacks.stacks.bionic io.paketo.stacks.tiny *]) (jre, 8.0.302, [io.buildpacks.stacks.bionic io.paketo.stacks.tiny *]) (jdk, 11.0.12, [io.buildpacks.stacks.bionic io.paketo.stacks.tiny *]) (jre, 11.0.12, [io.buildpacks.stacks.bionic io.paketo.stacks.tiny *]) (native-image-svm, 11.0.12, [io.buildpacks.stacks.bionic io.paketo.stacks.tiny *]) (jdk, 16.0.2, [io.buildpacks.stacks.bionic io.paketo.stacks.tiny *]) (jre, 16.0.2, [io.buildpacks.stacks.bionic io.paketo.stacks.tiny *])]
    [creator]     ERROR: failed to build: exit status 1

> Task :bootBuildImage FAILED
```

## Build Configuration

```
❯ java --version
openjdk 17 2021-09-14
OpenJDK Runtime Environment (build 17+35-Ubuntu-120.04)
OpenJDK 64-Bit Server VM (build 17+35-Ubuntu-120.04, mixed mode, sharing)
```

* What platform (`pack`, `kpack`, `tekton` buildpacks plugin, etc.) are you
  using? Please include a version.

* What buildpacks are you using? Please include versions.

* What builder are you using? If custom, can you provide the output from `pack
  inspect-builder <builder>`?

* Can you provide a sample app or relevant configuration (`buildpack.yml`,
  `nginx.conf`, etc.)?

Spring boot gradle plugin: `id("org.springframework.boot") version "2.5.4"`

`gradlew clean build bootBuildImage`

Similar issue was reported and solved for JDK 16: #86

**Comment by dmikusa-pivotal** [comment]
This should happen very soon. Oracle has released it, but we're still waiting on other vendors to cut releases. [Bellsoft Liberica](https://bell-sw.com/pages/downloads/?) is our default JVM vendor and at the moment I write this they have not released yet.

You can watch this issue for updates. I'll post back when we've got Java 17 releases from our upstream and we'll close out this issue when we've got buildpack releases available.

**Comment by dmikusa-pivotal** [comment]
We have https://github.com/paketo-buildpacks/amazon-corretto/releases/tag/v6.4.0 and https://github.com/paketo-buildpacks/sap-machine/releases/tag/v8.5.0 as the first two vendors to have releases published.

Bellsoft Liberica and Azul are close. We're monitoring those. Adoptium also has status updates available, so when they're ready we can pull that in too.

**Comment by dmikusa-pivotal** [comment]
Thanks @eddumelendez. I just merged in support for Java 17 in Bellsoft Liberica. The release has been published and paketo-buildpacks/java and the Paketo builders should update shortly (30-60 mins).

**Comment by linux-china** [comment]
It works now :) 

```
INFO]     [creator]       BellSoft Liberica JRE 17.0.0: Contributing to layer
[INFO]     [creator]         Downloading from https://github.com/bell-sw/Liberica/releases/download/17+35/bellsoft-jre17+35-linux-amd64.tar.gz
```

**Comment by dmikusa-pivotal** [comment]
Excellent. I'll leave this open & continue posting as we release Java 17 for the remainder of our Java Vendor buildpacks.

**Comment by dmikusa-pivotal** [comment]
Azul CNB has been updated to include Java 17. See https://github.com/paketo-buildpacks/azul-zulu/releases/tag/v8.6.0.

**Comment by Nikolas-Charalambidis** [comment]
@dmikusa-pivotal Any update on this? 😄

**Comment by dmikusa-pivotal** [comment]
Yes, sorry. I have missed a couple.

1. Microsoft OpenJDK released yesterday. We picked that up and a new version of those CNBs has been cut.
2. Adoptium also released last week. We picked that up and cut CNB releases as well.

The remaining vendors are:

- Alibaba Dragonwell 17
- Eclipse OpenJ9 / IBM Semeru 17
- GraalVM 17

We'll continue monitoring those and cut releases as soon as upstream downloads are available.
